### DISCUSSION TITLE
Same CSS Selector can have three values - how to handle?

### DISCUSSION THREAD

**Discussion by alpha-code2019** [AutomaApp/automa#522] [Q&A]
Hello Kholid,

sorry, but I am stuck with the following problem. Please try to help me. I want to understand your Automa app much better.

In the web page I want to scrape, with the help of the app selector  I found two CSS selectors

Data Field Name:
**.odd:nth-child(2) > .title**

Data Field Value:
**.odd:nth-child(2) > b**

The problem is that the Data Field Name CSS Selector can be one of three possible values:

Real Estate
Rent
Established

Since I have to name the scraper field, only 1 out of three instances is true.
I have to name the fields so that they can be sent to the correct column in the result table.

Table Column 1:	Real Estate
Table Column 2:	Rent
Table Column 3:	Established

How could this complex problem be solved so that Automa scrapes it as needed?

**Comment by Kholid060** [comment]
For this problem, I guess you can use the Conditions block to check whether the text is "Real Estate", "Rent", or "Established". And when one of those conditions matches, it will insert it into column 1, column 2, or column 3.

![image](https://user-images.githubusercontent.com/22908993/171801044-4bc66a81-0e48-4c7f-8bfa-e805cc331cfb.png)
![image](https://user-images.githubusercontent.com/22908993/171801753-56056a9b-15e6-4262-9d1b-da0d1069f47f.png)

Or you can use the javascript code block.

```js
const element = document.querySelector('.odd:nth-child(2) > b');
const elementText = element.innerText; // 'Real Estate', 'Rent', or 'Established'

const columnsName = {
  'Real Estate': 'Column 1',
  'Rent': 'Column 2',
  'Established': 'Column 3',
}
const column = columnsName[elementText];

automaNextBlock({ [column]: elementText })
```

**Comment by alpha-code2019** [comment]
Wow, thank you so much!

BTW: Is it OK if I call you Ahmad?

Am Fr., 3. Juni 2022 um 08:54 Uhr schrieb Ahmad Kholid <
***@***.***>:

**Reply by Kholid060** [reply]
It's okay

**Comment by alpha-code2019** [comment]
FYI:

I have published two test script, so you can better check, what I am doing wrong

BisBuySell Test 01
https://www.automa.site/workflow/pVVvqNowZWFMAOpYJlbLz

BizBuySell Test 02 (JavaScript)
https://www.automa.site/workflow/v6m8Oq65NbdJVAPPtdVtg

**Comment by Kholid060** [comment]
I think I manage to make it work, you can try it by importing the below workflow but you have to change the file extension from `.txt` to `.json` first.
[BizBuySell.txt](https://github.com/AutomaApp/automa/files/8836013/BizBuySell.txt)

The way it works is using the XPath selector in the Get Text block like 
`//span[contains(child::text(), "Real Estate")]/following-sibling::*`
The selector will try to find a `span` element where the text contains the `Real Estate` word and select any element next to it.

**Comment by alpha-code2019** [comment]


**Reply by alpha-code2019** [reply]
Hi Ahmad,

I have found the correct XML selector to scrape it:

//dl[@id="ctl00_ctl00_Content_ContentPlaceHolder1_wideProfile_listingDetails_dlDetailedInformation"]/dt[. = "Location:"]/following-sibling::dd

Thank you!
